python/cuml/linear_model/linear_regression_mg.pyx

```diff
@@ -23,610 +22,474 @@ import ctypes
+        cdef uintptr_t data_ptr = data
+        cdef doubleData_t **d = <doubleData_t**>data_ptr
+        for x_i in range(len(arr_interfaces)):
+            free(d[x_i])
+        free(d)
+
+    def _freeFloatD(self, data, arr_interfaces):
+        cdef uintptr_t data_ptr = data
+        cdef floatData_t **d = <floatData_t**>data_ptr
+        for x_i in range(len(arr_interfaces)):
+            free(d[x_i])
+        free(d)
+
+    def _build_predData(self, partsToRanks, rnk, n_cols, dtype):
+        arr_interfaces_trans = []
+        for idx, rankSize in enumerate(partsToRanks):
+            rank, size = rankSize
+            if rnk == rank:
+                trans_ary = rmm.to_device(zeros((size, n_cols),
+                                                order="F",
+                                                dtype=dtype))
+
+                trans_ptr = get_dev_array_ptr(trans_ary)
+                arr_interfaces_trans.append({"obj": trans_ary,
+                                             "data": trans_ptr,
+                                             "shape": (size, n_cols)})
+
+        return arr_interfaces_trans
+
+    def fit(self, X, y, M, N, partsToRanks, rnk):
+        arr_interfaces = []
+        for arr in X:
+            X_m, input_ptr, n_rows, self.n_cols, self.dtype = \
+                input_to_dev_array(arr, check_dtype=[np.float32, np.float64])
+            arr_interfaces.append({"obj": X_m,
+                                   "data": input_ptr,
+                                   "shape": (n_rows, self.n_cols)})
+
+        arr_interfaces_y = []
+        for arr in y:
+            y_m, input_ptr, n_rows, n_cols, self.dtype = \
+                input_to_dev_array(arr, check_dtype=[np.float32, np.float64])
+            arr_interfaces_y.append({"obj": y_m,
+                                     "data": input_ptr,
+                                     "shape": (n_rows, n_cols)})
+
+        n_total_parts = 0
+        for idx, rankSize in enumerate(partsToRanks):
+            rank, size = rankSize
+            if rnk == rank:
+                n_total_parts = n_total_parts + 1
+
+        cdef RankSizePair **rankSizePair = <RankSizePair**> \
+            malloc(sizeof(RankSizePair**)
+                   * n_total_parts)
+
+        indx = 0
+        n_part_row = 0
+
+        for idx, rankSize in enumerate(partsToRanks):
+            rank, size = rankSize
+            if rnk == rank:
+                rankSizePair[indx] = <RankSizePair*> \
+                    malloc(sizeof(RankSizePair))
+                rankSizePair[indx].rank = <int>rank
+                rankSizePair[indx].size = <size_t>size
+                n_part_row = n_part_row + rankSizePair[indx].size
+                indx = indx + 1
+
+        self.coef_ = cudf.Series(zeros(self.n_cols,
+                                       dtype=self.dtype))
+        cdef uintptr_t coef_ptr = get_cudf_column_ptr(self.coef_)
+
+        cdef float c_intercept1
+        cdef double c_intercept2
```

Comment:
why are they named c_intercept1 and c_intercept2? do they differ only in datatype? Would be nice to call that out if so

Reply:
Done.